pkg/labels/labels.go

```diff
@@ -257,6 +257,25 @@ func (l *Label) String() string {
 	return fmt.Sprintf("%s:%s", l.Source, l.Key)
 }
 
+func (l *Label) StringConcat() string {
+	if len(l.Value) != 0 {
+		return l.Source + ":" + l.Key + "=" + l.Value
+	}
+	return l.Source + ":" + l.Key
+}
+
+func (l *Label) StringBuilder() string {
+	var b strings.Builder
```

Comment:
Can you try putting the `strings.Builder` object onto the Label so it's not created every time? We can just call `Reset()` in this function.

Reply:
not much difference 🤔 
```
goos: darwin
goarch: amd64
pkg: github.com/cilium/cilium/pkg/labels

BenchmarkLabel_String
BenchmarkLabel_String-8      5469518         216 ns/op

BenchmarkLabel_StringConcat
BenchmarkLabel_StringConcat-8     19706568          59.7 ns/op

BenchmarkLabel_StringBuilder
BenchmarkLabel_StringBuilder-8    11228988         101 ns/op
```

Reply:
@glibsm the issue with the way i used strings.Builder was that there were multiple memory allocations because Reset() sets the underlying buffer to nil: https://golang.org/src/strings/builder.go?s=1869:1894#L50

i added a call to Grow() so that there is a single memory allocation, and it performed better than string concatenation version:
```
func (l *Label) String() string {
	// With strings.Builder being a part of the Label struct.
	l.builder.Reset()
	l.builder.Grow(len(l.Source) + len(l.Key) + len(l.Value) + 2)
	l.builder.WriteString(l.Source)
	l.builder.WriteString(":")
	l.builder.WriteString(l.Key)
	if len(l.Value) != 0 {
		l.builder.WriteString("=")
		l.builder.WriteString(l.Value)
	}
	return l.builder.String()
}

BenchmarkLabel_String-8   	25152483	        47.7 ns/op	      48 B/op	       1 allocs/op
```
i'm still leaning towards sticking with the string concatenation version. it performs well enough, and looks much simpler. what do you think?

Reply:
Concat seems good enough for now. If it's still a problem in future flame graphs, we can come back and re-visit the builder. Biggest gain is probably going to be from memoizing the labels and not creating them each time would be my guess but lets see what pprof says.